src/Couchbase.Lite.Shared/Document/InMemoryDictionary.cs

```diff
@@ -308,6 +310,12 @@ public IMutableDictionary SetValue(string key, object value)
             return this;
         }
 
+        public IMutableDictionary SetJSON([NotNull] string json)
+        {
+            var blobDict = JsonConvert.DeserializeObject<Dictionary<string, object>>(json);
```

Comment:
exception handling?

Reply:
JsonConvert.DeserializeObject will throw json string is unable to deserialize into a dictionary type.

Reply:
That's the point, it should be wrapped in a CBL exception, with this one as the inner one.

Reply:
done